lib/api/objectCopy.js

```diff
@@ -140,28 +141,46 @@ function objectCopy(authInfo, request, sourceBucket,
     }
 
     return async.waterfall([
-        function checkSourceAuthorization(next) {
+        function checkDestAuth(next) {
+            return services.metadataValidateAuthorization(valPutParams,
```

Comment:
Same here, why is the check moved to be the first step?

Edit: I see that you need destBucketMD for the error callbacks. I would suggest on calling the metadataGetBucket directly and process the checks in `checkDestAuth` by removing services.metadataValidateAuthorization directly.